### PR TITLE
[WIP] Replace Rx

### PR DESCRIPTION
**Pull request by SeanFeldman**
RX was throwing NRE exception.
@yvesgoeleven have a look at this approach.

### CODE REVIEW

File: samples/azure/performance-tuning-asb/ASB_7/Shared/ReceiveCounter.cs [hunk 1]
```diff
@@ -1,23 +1,34 @@
 ﻿using System;
-using System.Reactive.Linq;
-using System.Reactive.Subjects;
+using System.Threading;
+using System.Timers;
```
Comment (indualagarsamy) [inline samples/azure/performance-tuning-asb/ASB_7/Shared/ReceiveCounter.cs]: Why System.Timers and not System.Threading.Timers?

Comment (SeanFeldman) [inline samples/azure/performance-tuning-asb/ASB_7/Shared/ReceiveCounter.cs]: I recall a difference between the two, where Threading.Timers was not advised.

Comment (SimonCropp) [inline samples/azure/performance-tuning-asb/ASB_7/Shared/ReceiveCounter.cs]: my understanding is System.Timers.Timer is primarily for a design surface. so in this case we should use System.Threading.Timers. System.Threading.Timers is what we use in the core

Comment (SeanFeldman) [inline samples/azure/performance-tuning-asb/ASB_7/Shared/ReceiveCounter.cs]: Ah, it's the other way around. Thanks @SimonCropp, fixed.

**Comment by SeanFeldman** [comment]
Long ago had the same question and was quite surprised:
http://stackoverflow.com/a/1416809/59563

**Comment by indualagarsamy** [comment]
@seanfeldman:

System.Threading.Timer is a plain timer. It calls you back on a thread pool thread (from the worker pool).

System.Timers.Timer is a System.ComponentModel.Component that wraps a System.Threading.Timer, and provides some additional features used for dispatching on a particular thread.

System.Windows.Forms.Timer instead wraps a native message-only-HWND and uses Window Timers to raise events in that HWNDs message loop.

If your app has no UI, and you want the most light-weight and general-purpose .Net timer possible, (because you are happy figuring out your own threading/dispatching) then System.Threading.Timer is as good as it gets in the framework.



From: http://stackoverflow.com/questions/1416803/system-timers-timer-vs-system-threading-timer

**Comment by SeanFeldman** [comment]
@indualagarsamy it has been already updated :)

**Comment by SimonCropp** [comment]
@SeanFeldman should this still have a WIP?

**Comment by SeanFeldman** [comment]
@SimonCropp I was hoping @yvesgoeleven could have a look since I'm taking out Rx he has implemented originally. If you're happy with implementation, please merge.

**Comment by SeanFeldman** [comment]
Superseded by #2128

**Pull request closed without merging**